File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        def bfs(i,adj,visited):
            visited[i]=True
            q=deque()
            q.append(i)
            while len(q)!=0:
                num=q.popleft()
                for i in range(len(adj)):
                    if adj[num][i]==1 and num!=i and visited[i]==False:
                        q.append(i)
                        visited[i]=True
        adj=isConnected
        n=len(adj)
        visited=[False for i in range(n)]
        ans=0
        for i in range(n):
            if visited[i]==False:
                ans+=1
                bfs(i,adj,visited)
        return ans

                                                                                                                                                                                                                                                                            

File: 0547-number-of-provinces/0547-number-of-provinces.java
class Solution {
    public void bfs(int node,int[][] isConnected,boolean[] visited){
        Queue<Integer> q=new LinkedList<>();
        q.add(node);
        visited[node]=true;
        while(!q.isEmpty()){
            int curr=q.poll();
            for(int i=0;i<isConnected[curr].length;i++){
                if(isConnected[curr][i]==1 && visited[i]==false){
                    visited[i]=true;
                    q.add(i);
                }
            }
        }
    }
    public int findCircleNum(int[][] isConnected) {
        int n=isConnected.length;
        boolean[] visited=new boolean[n];
        int ans=0;
        for(int i=0;i<n;i++){
            if(!visited[i]){
                bfs(i,isConnected,visited);
                ans++;
            }
        }
        return ans;
    }
}